src/os/bluestore/BlueStore.cc

```diff
@@ -564,20 +564,33 @@ static int get_key_pool_stat(const string& key, uint64_t* pool_id)
 }
 
 #ifdef HAVE_LIBZBD
-static void get_zone_offset_key(uint32_t zone, uint64_t offset, std::string *key)
```

Comment:
The idea was to make the references sorted, so that we would process objects to clean in (approximate) order they appear on disk.  In retrospect, though, I think we should consider just reading the whole zone into memory in one go, or possibly use a granular freelist so we read just the live extents.  Even if we read a bunch of dead data, on a spinning disk that is likely to be way more efficient than seeking at least once for each live object.  Or perhaps scan the set of live objects to build the live extent list, read them, then scan again to rewrite and update the object metadata.  None of those options need the offset.